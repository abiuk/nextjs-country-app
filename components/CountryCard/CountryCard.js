import React from "react";
import styled from "styled-components";
import Link from "next/link";

const Root = styled.div`
  display: flex;
  flex-direction: column;

  @media screen and (min-width: 600px) {
    flex-direction: row;
    flex-wrap: wrap;
  }
`;

const Card = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 14px;
  border: 1px solid #e9e9e9;
  cursor: pointer;

  color: inherit;
  text-decoration: none;

  @media screen and (min-width: 600px) {
    flex-direction: row;
    flex: 1;
    margin: 14px;

    transition: 0.5s ease;

    :hover {
      transform: scale(1.05);
      box-shadow: 5px 5px 15px rgba(0, 0, 0, 0.6);
    }
  }
`;

const AElement = styled.a`
  display: flex;
  flex-direction: column;

  @media screen and (min-width: 600px) {
    flex-direction: row;
  }
`;

const ImageColumn = styled.div`
  margin-right: 16px;
`;

export const ImgElement = styled.img`
  width: 100%;

  @media screen and (min-width: 600px) {
    width: 220px;
    height: 150px;
  }
`;

const InfoColumn = styled.div`
  flex-grow: 1;
  margin: auto;
`;

export const InfoRow = styled.div`
  display: flex;
  margin-bottom: 16px;
`;

export const Info = styled.div`
  font-weight: 600;
  min-width: 90px;
  margin-right: 24px;
`;

export const Value = styled.div`
  color: #525252;
`;

const NO_CAPITAL = "N/A";

const CountryCard = ({ countries }) => (
  <Root>
    {countries.map((country) => (
      <Card key={country.alpha3Code}>
        <Link
          href={{
            pathname: "/country/[code]",
            query: { code: country.alpha3Code },
          }}
        >
          <AElement>
            <ImageColumn>
              <ImgElement src={country.flag} />
            </ImageColumn>
            <InfoColumn>
              <InfoRow>
                <Info>Name</Info>
                <Value>{country.name}</Value>
              </InfoRow>
              <InfoRow>
                <Info>Population</Info>
                <Value>{country.population.toLocaleString()}</Value>
              </InfoRow>
              <InfoRow>
                <Info>Capital</Info>
                <Value>{country.capital || NO_CAPITAL}</Value>
              </InfoRow>
            </InfoColumn>
          </AElement>
        </Link>
      </Card>
    ))}
  </Root>
);

export default CountryCard;
