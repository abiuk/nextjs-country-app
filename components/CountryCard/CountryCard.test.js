import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

import React from "react";
import renderer from "react-test-renderer";

import CountryCard from "./CountryCard";

jest.mock("next/router", () => ({ push: jest.fn() }));

const countries = [
  {
    name: "Fiji",
    topLevelDomain: [".fj"],
    alpha2Code: "FJ",
    alpha3Code: "FJI",
    callingCodes: ["679"],
    capital: "Suva",
    altSpellings: [
      "FJ",
      "Viti",
      "Republic of Fiji",
      "Matanitu ko Viti",
      "Fijī Gaṇarājya",
    ],
    region: "Oceania",
    subregion: "Melanesia",
    population: 867000,
    latlng: [-18, 175],
    demonym: "Fijian",
    area: 18272,
    gini: 42.8,
    timezones: ["UTC+12:00"],
    borders: [],
    nativeName: "Fiji",
    numericCode: "242",
    currencies: [[Object]],
    languages: [[Object], [Object], [Object], [Object]],
    translations: {
      de: "Fidschi",
      es: "Fiyi",
      fr: "Fidji",
      ja: "フィジー",
      it: "Figi",
      br: "Fiji",
      pt: "Fiji",
      nl: "Fiji",
      hr: "Fiđi",
      fa: "فیجی",
    },
    flag: "https://restcountries.eu/data/fji.svg",
    regionalBlocs: [],
    cioc: "FIJ",
  },
  {
    name: "Finland",
    topLevelDomain: [".fi"],
    alpha2Code: "FI",
    alpha3Code: "FIN",
    callingCodes: ["358"],
    capital: "Helsinki",
    altSpellings: [
      "FI",
      "Suomi",
      "Republic of Finland",
      "Suomen tasavalta",
      "Republiken Finland",
    ],
    region: "Europe",
    subregion: "Northern Europe",
    population: 5491817,
    latlng: [64, 26],
    demonym: "Finnish",
    area: 338424,
    gini: 26.9,
    timezones: ["UTC+02:00"],
    borders: ["NOR", "SWE", "RUS"],
    nativeName: "Suomi",
    numericCode: "246",
    currencies: [[Object]],
    languages: [[Object], [Object]],
    translations: {
      de: "Finnland",
      es: "Finlandia",
      fr: "Finlande",
      ja: "フィンランド",
      it: "Finlandia",
      br: "Finlândia",
      pt: "Finlândia",
      nl: "Finland",
      hr: "Finska",
      fa: "فنلاند",
    },
    flag: "https://restcountries.eu/data/fin.svg",
    regionalBlocs: [[Object]],
    cioc: "FIN",
  },
  {
    name: "France",
    topLevelDomain: [".fr"],
    alpha2Code: "FR",
    alpha3Code: "FRA",
    callingCodes: ["33"],
    capital: "Paris",
    altSpellings: ["FR", "French Republic", "République française"],
    region: "Europe",
    subregion: "Western Europe",
    population: 66710000,
    latlng: [46, 2],
    demonym: "French",
    area: 640679,
    gini: 32.7,
    timezones: [
      "UTC-10:00",
      "UTC-09:30",
      "UTC-09:00",
      "UTC-08:00",
      "UTC-04:00",
      "UTC-03:00",
      "UTC+01:00",
      "UTC+03:00",
      "UTC+04:00",
      "UTC+05:00",
      "UTC+11:00",
      "UTC+12:00",
    ],
    borders: ["AND", "BEL", "DEU", "ITA", "LUX", "MCO", "ESP", "CHE"],
    nativeName: "France",
    numericCode: "250",
    currencies: [[Object]],
    languages: [[Object]],
    translations: {
      de: "Frankreich",
      es: "Francia",
      fr: "France",
      ja: "フランス",
      it: "Francia",
      br: "França",
      pt: "França",
      nl: "Frankrijk",
      hr: "Francuska",
      fa: "فرانسه",
    },
    flag: "https://restcountries.eu/data/fra.svg",
    regionalBlocs: [[Object]],
    cioc: "FRA",
  },
];

describe("CountryCard", () => {
  it("should render successfully", () => {
    const tree = renderer
      .create(<CountryCard countries={countries} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
