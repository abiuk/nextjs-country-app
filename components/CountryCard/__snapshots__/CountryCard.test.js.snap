// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryCard should render successfully 1`] = `
<div
  className="CountryCard__Root-sc-187yvkl-0 kfHFOy"
>
  <div
    className="CountryCard__Card-sc-187yvkl-1 eDcFFh"
  >
    <a
      className="CountryCard__AElement-sc-187yvkl-2 iSGwnK"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="CountryCard__ImageColumn-sc-187yvkl-3 fZfVFV"
      >
        <img
          className="CountryCard__ImgElement-sc-187yvkl-4 gRLlmY"
          src="https://restcountries.eu/data/fji.svg"
        />
      </div>
      <div
        className="CountryCard__InfoColumn-sc-187yvkl-5 cnoopI"
      >
        <div
          className="CountryCard__InfoRow-sc-187yvkl-6 hxqtoq"
        >
          <div
            className="CountryCard__Info-sc-187yvkl-7 ibXQgH"
          >
            Name
          </div>
          <div
            className="CountryCard__Value-sc-187yvkl-8 eLvmaF"
          >
            Fiji
          </div>
        </div>
        <div
          className="CountryCard__InfoRow-sc-187yvkl-6 hxqtoq"
        >
          <div
            className="CountryCard__Info-sc-187yvkl-7 ibXQgH"
          >
            Population
          </div>
          <div
            className="CountryCard__Value-sc-187yvkl-8 eLvmaF"
          >
            867,000
          </div>
        </div>
        <div
          className="CountryCard__InfoRow-sc-187yvkl-6 hxqtoq"
        >
          <div
            className="CountryCard__Info-sc-187yvkl-7 ibXQgH"
          >
            Capital
          </div>
          <div
            className="CountryCard__Value-sc-187yvkl-8 eLvmaF"
          >
            Suva
          </div>
        </div>
      </div>
    </a>
  </div>
  <div
    className="CountryCard__Card-sc-187yvkl-1 eDcFFh"
  >
    <a
      className="CountryCard__AElement-sc-187yvkl-2 iSGwnK"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="CountryCard__ImageColumn-sc-187yvkl-3 fZfVFV"
      >
        <img
          className="CountryCard__ImgElement-sc-187yvkl-4 gRLlmY"
          src="https://restcountries.eu/data/fin.svg"
        />
      </div>
      <div
        className="CountryCard__InfoColumn-sc-187yvkl-5 cnoopI"
      >
        <div
          className="CountryCard__InfoRow-sc-187yvkl-6 hxqtoq"
        >
          <div
            className="CountryCard__Info-sc-187yvkl-7 ibXQgH"
          >
            Name
          </div>
          <div
            className="CountryCard__Value-sc-187yvkl-8 eLvmaF"
          >
            Finland
          </div>
        </div>
        <div
          className="CountryCard__InfoRow-sc-187yvkl-6 hxqtoq"
        >
          <div
            className="CountryCard__Info-sc-187yvkl-7 ibXQgH"
          >
            Population
          </div>
          <div
            className="CountryCard__Value-sc-187yvkl-8 eLvmaF"
          >
            5,491,817
          </div>
        </div>
        <div
          className="CountryCard__InfoRow-sc-187yvkl-6 hxqtoq"
        >
          <div
            className="CountryCard__Info-sc-187yvkl-7 ibXQgH"
          >
            Capital
          </div>
          <div
            className="CountryCard__Value-sc-187yvkl-8 eLvmaF"
          >
            Helsinki
          </div>
        </div>
      </div>
    </a>
  </div>
  <div
    className="CountryCard__Card-sc-187yvkl-1 eDcFFh"
  >
    <a
      className="CountryCard__AElement-sc-187yvkl-2 iSGwnK"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="CountryCard__ImageColumn-sc-187yvkl-3 fZfVFV"
      >
        <img
          className="CountryCard__ImgElement-sc-187yvkl-4 gRLlmY"
          src="https://restcountries.eu/data/fra.svg"
        />
      </div>
      <div
        className="CountryCard__InfoColumn-sc-187yvkl-5 cnoopI"
      >
        <div
          className="CountryCard__InfoRow-sc-187yvkl-6 hxqtoq"
        >
          <div
            className="CountryCard__Info-sc-187yvkl-7 ibXQgH"
          >
            Name
          </div>
          <div
            className="CountryCard__Value-sc-187yvkl-8 eLvmaF"
          >
            France
          </div>
        </div>
        <div
          className="CountryCard__InfoRow-sc-187yvkl-6 hxqtoq"
        >
          <div
            className="CountryCard__Info-sc-187yvkl-7 ibXQgH"
          >
            Population
          </div>
          <div
            className="CountryCard__Value-sc-187yvkl-8 eLvmaF"
          >
            66,710,000
          </div>
        </div>
        <div
          className="CountryCard__InfoRow-sc-187yvkl-6 hxqtoq"
        >
          <div
            className="CountryCard__Info-sc-187yvkl-7 ibXQgH"
          >
            Capital
          </div>
          <div
            className="CountryCard__Value-sc-187yvkl-8 eLvmaF"
          >
            Paris
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
